### PR TITLE
Update docs / change Spectral Norm

### PR DESCRIPTION
**Pull request by charlie-becker**
Updated doc strings for `layers.py` and `losses.py` and changed the Spectral normalization in the regression layers to be optional (defaulted to `False`).

### CODE REVIEW

File: mlguess/keras/layers.py [hunk 1]
```diff
@@ -1,23 +1,32 @@
...
-    
+    """
+    Implements dense layer for an deep evidential regression model.
     Reference: https://www.mit.edu/~amini/pubs/pdf/deep-evidential-regression.pdf
     Source: https://github.com/aamini/evidential-deep-learning
+
+    Args:
+        units (int): Output size of regression tasks
+        name (str): Name of the layer
+        spectral_normalization (Bool): Whether to use Spectral Normalization
```
Comment (djgagne) [inline mlguess/keras/layers.py:13]: Bool should not be capitalized.

File: mlguess/keras/layers.py [hunk 2]
```diff
@@ -42,11 +51,25 @@ def get_config(self):
         
         
 class DenseNormal(tf.keras.layers.Layer):
-    def __init__(self, units, name, eps=1e-12):
+    """ Dense layer for Gaussian distribution regression.
+    Args:
+        units (int): Output size of regression tasks
+        name (str): Name of the layer
+        spectral_normalization (Bool): Whether to use Spectral Normalization
```
Comment (djgagne) [inline mlguess/keras/layers.py:58]: Bool should not be capitalized.

File: mlguess/keras/losses.py [hunk 3]
```diff
@@ -124,13 +111,19 @@ def get_config(self):
 
 
 def GaussianNLL(y, y_pred, reduce=True):
```
Comment (djgagne) [inline mlguess/keras/losses.py:113]: we should change the name of this function to be lowercase and separated by underscores: `gaussian_nll`

File: mlguess/keras/losses.py [hunk 4]
```diff
@@ -150,32 +143,24 @@ def NIG_NLL(self, y, gamma, v, alpha, beta, reduce=True):
...
-        )
+        nll = (0.5 * tf.math.log(np.pi / v)
+               - alpha * tf.math.log(twoBlambda)
+               + (alpha + 0.5) * tf.math.log(v * (y - gamma) ** 2 + twoBlambda)
+               + tf.math.lgamma(alpha)
+               - tf.math.lgamma(alpha + 0.5))
 
         return tf.reduce_mean(nll) if reduce else nll
 
     def NIG_Reg(self, y, gamma, v, alpha, reduce=True):
```
Comment (djgagne) [inline mlguess/keras/losses.py:154]: should be renamed to be clearer.

**Review by djgagne — CHANGES_REQUESTED** [review]
I have a couple of small doc changes but otherwise it looks good.

**Review by djgagne — APPROVED** [review]
Looks good to me.
